### DISCUSSION TITLE
Change Restore method to be a "Safe Restore"

### DISCUSSION THREAD

**Discussion by DantasB** [Altinity/clickhouse-backup#936] [Ideas]
It's just a suggestion to the clickhouse-backup CLI.

When you apply a restore, passing the --schema and --data, the steps are basically:
1. Drop the table if exists
2. Create table
3. Attach partitions to this existing table.

But, lets supose that the attach partitions, for some how, just breaks? For that, i would suggest to do a:
1. Drop the table`_tmp` if exists
2. Create table`_tmp`
3. Attach partitions to this existing table`_tmp`
4. Drop the table if exists
5. Rename table`_tmp` to table.

With that approach, the system would be almost always up and could be safer. And if any problem occurs in the tmp table, it would be just a tmp table and could not bring ANY data loss, or am I wrong?

**Comment by Slach** [comment]
this approach have side effect and need a lot of additional logic related to Materialized View, Replicated* and distributed engines

**Reply by DantasB** [reply]
OOOF, what kind of side effect? Just curious about it

**Reply by Slach** [reply]
lot of corner cases with replication_path parameter
need lot of work with dependencies (for example dictionary with SOURCE(CLICKHOUSE(...)))
